src/stage1/compiler_utils.jl

```diff
@@ -64,3 +64,29 @@ Base.lastindex(x::Core.Compiler.InstructionStream) =
 if isdefined(Core.Compiler, :CallInfo)
     Base.convert(::Type{Core.Compiler.CallInfo}, ::Nothing) = Core.Compiler.NoCallInfo()
 end
+
+
+"""
+    find_end_of_phi_block(ir, start_search_idx)
+
+Finds the last index within the same basic block, on or after the `start_search_idx` which is not within a phi block.
+A phi-block is a run on PhiNodes or nothings that must be the first statements within the basic block.
+
+If `start_search_idx` is not within a phi block to begin with, then just returns `start_search_idx`
+"""
+function find_end_of_phi_block(ir, start_search_idx::Int)
+    # Short-cut for early exit:
+    stmt = ir.stmts[start_search_idx][:inst]
+    stmt !== nothing && !isa(stmt, PhiNode) && return start_search_idx
+
+    # Actually going to have to go digging throught the IR to out if were are in a phi block
+    # TODO: this is not so efficient. maybe preconstruct CFG then use block_for_inst?
+    bb=CC.block_for_inst(ir.cfg, start_search_idx)
+    end_search_idx=ir.cfg.blocks[bb].stmts[end]
+    for idx in (start_search_idx+1):(end_search_idx-1)
```

Comment:
This is possibly off by one? What if `start_search_idx` is the last PhiNode?

Reply:
Ah yes, I thought we had already checked that one, but because this search is checking `idx+1` we haven't.
```suggestion
    for idx in (start_search_idx):(end_search_idx-1)
```